lms/resources/lti_launch.py

```diff
@@ -82,6 +82,19 @@ def h_group(self):
     def _is_speed_grader(self):
         return bool(self._request.GET.get("learner_canvas_user_id"))
 
+    @property
+    def is_legacy_speed_grader(self):
+        # Detect speed grader launches without the fix for
+        #
+        # * https://github.com/instructure/canvas-lms/issues/1952
+        # * https://github.com/hypothesis/lms/issues/3228
+        #
+        # applied which includes the correct value for `resource_link_id` in the query params.
+        return (
+            self._is_speed_grader
+            and self.resource_link_id == self._request.POST["context_id"]
```

Comment:
I wonder if we want something like this instead?

```suggestion
            and "resource_link_id" not in self._request.GET
```

I think a legacy SpeedGrader URL is one that we submitted to Canvas before we started adding `resource_link_id` query params to SpeedGrader URLs. If we instead rely on `self.resource_link_id == self._request.POST["context_id"]` then this might start returning `False` instead of `True` if Instructure fix https://github.com/instructure/canvas-lms/issues/1952.